includes/class-affiliatewp-allow-own-referrals.php

```diff
@@ -0,0 +1,328 @@
+				load_textdomain( 'affiliatewp-allow-own-referrals', $mofile_local );
+
+			} else {
+
+				// Load the default language files.
+				load_plugin_textdomain( 'affiliatewp-allow-own-referrals', false, $lang_dir );
+
+			}
+		}
+
+		/**
+		 * Includes necessary files.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		private function includes() {
+			// Bring in the autoloader.
+			require_once dirname( __FILE__ ) . '/lib/autoload.php';
+
+			// require_once AFFWP_AOR_PLUGIN_DIR . 'includes/file-name.php';
+		}
+
+		/**
+		 * Sets up the default hooks and actions.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @return void
+		 */
+		private function hooks() {
+			// Plugin meta.
+			add_filter( 'plugin_row_meta', array( $this, 'plugin_meta' ), null, 2 );
+			add_filter( 'affwp_is_customer_email_affiliate_email', '__return_false' );
+			add_filter( 'affwp_tracking_is_valid_affiliate', [ $this, 'tracking_override' ], 10, 2 );
+		}
+
+		/**
+		 * Overrides valid affiliate tracking to make it possible for affiliates to refer themselves.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @param bool $valid        The current filtered status of this affiliate's validity.
+		 * @param int  $affiliate_id The affiliate ID to check against.
+		 * @return bool True if the affiliate is valid, otherwise false.
+		 */
+		public function tracking_override( $valid, $affiliate_id ) {
+
+			if ( 'active' === affwp_get_affiliate_status( $affiliate_id ) ) {
+				$valid = true;
+			}
+
+			return $valid;
+		}
+
+		/**
+		 * Modifies the plugin list table meta links.
+		 *
+		 * @since 1.0.3
+		 *
+		 * @param array  $links The current links array.
+		 * @param string $file  A specific plugin table entry.
+		 * @return array The modified links array.
+		 */
+		public function plugin_meta( $links, $file ) {
+
+			if ( $file == plugin_basename( __FILE__ ) ) {
+
+				$url = admin_url( 'admin.php?page=affiliate-wp-add-ons' );
+
+				$plugins_link = array( '<a alt="' . esc_attr__( 'Get more add-ons for AffiliateWP', 'affiliatewp-allow-own-referrals' ) . '" href="' . esc_url( $url ) . '">' . __( 'More add-ons', 'affiliatewp-allow-own-referrals' ) . '</a>' );
+
+				$links = array_merge( $links, $plugins_link );
+			}
+
+			return $links;
+
+		}
+	}
+}
+
+/**
+ * The main function responsible for returning the one true bootstrap instance
+ * to functions everywhere.
+ *
+ * Use this function like you would a global variable, except without needing
+ * to declare the global.
+ *
+ * Example: <?php $affiliatewp_allow_own_referrals = affiliatewp_allow_own_referrals(); ?>
+ *
+ * @since 1.0.3
+ *
+ * @return \AffiliateWP_Allow_Own_Referrals The one true bootstrap instance.
+ */
+function affiliatewp_allow_own_referrals() {
+	if ( ! class_exists( 'Affiliate_WP' ) ) {
+
+		if ( ! class_exists( 'AffiliateWP_Activation' ) ) {
+			require_once 'includes/class-activation.php';
+		}
+
+		$activation = new AffiliateWP_Activation( plugin_dir_path( __FILE__ ), basename( __FILE__ ) );
```

Comment:
`__FILE__` should be `$this->file` pointing to the main plugin file.
```suggestion
		$activation = new AffiliateWP_Activation( plugin_dir_path( $this->file ), basename( $this->file ) );
```